autokeras/hypermodel/preprocessor.py

```diff
@@ -694,11 +696,14 @@ class FeatureEngineering(Preprocessor):
             Defaults to 1000.
     """
 
-    def __init__(self, column_types, max_columns=1000, **kwargs):
+    def __init__(self, input_node, max_columns=1000, **kwargs):
+        # TODO: support partial column_types, i.e., the size of the dict is smaller
+        # than the number of the columns.
         super().__init__(**kwargs)
-        self.column_types = column_types
+        self.column_types = input_node.column_types
```

Comment:
use self.input_node instead of directly use self.column_types or self.column_names.